api/coreservice.go

```diff
@@ -491,7 +491,21 @@ func (core *coreService) SendAction(ctx context.Context, in *iotextypes.Action)
 }
 
 func (core *coreService) PendingNonce(addr address.Address) (uint64, error) {
-	return core.ap.GetPendingNonce(addr.String())
+	nonce, err := core.ap.GetPendingNonce(addr.String())
+	if err != nil {
+		return nonce, err
+	}
+	g := core.Genesis()
+	if nonce > 1 || !g.IsSumatra(core.TipHeight()) {
```

Comment:
why `IsSumatra`? what's the difference from `UseZeroNonceForFreshAccount`?

Reply:
`IsSumatra()` is the same flag as `UseZeroNonceForFreshAccount `, here we don't have a `ctx`

API service does not affect past blocks, so after the hard-fork, this check will be removed

Reply:
if so, be consistent with `g.IsToBeEnabled(core.TipHeight())`